### PR TITLE
Add dropout to DeepMIL and fix feature extractor setup

### PR DESCRIPTION
**Pull request by dccastro**

### CODE REVIEW

File: InnerEye/ML/Histopathology/models/deepmil.py [hunk 1]
```diff
@@ -115,6 +119,11 @@ def get_pooling(self) -> Tuple[Callable, int]:
         num_features = self.num_encoding*self.pool_out_dim
         return pooling_layer, num_features
 
+    def get_dropout(self) -> Callable:
```
Comment (harshita-s) [inline InnerEye/ML/Histopathology/models/deepmil.py:122]: Maybe we could also add a function for hidden layers? We could add an optional hidden layer before the dropout and classifier in `forward()`

File: InnerEye/ML/Histopathology/utils/layer_utils.py [hunk 2]
```diff
@@ -16,14 +16,22 @@ def get_imagenet_preprocessing() -> nn.Module:
 
 def setup_feature_extractor(pretrained_model: nn.Module,
                             input_dim: Tuple[int, int, int]) -> Tuple[Callable, int]:
-    layers = list(pretrained_model.children())[:-1]
-    layers.append(nn.Flatten())  # flatten non-batch dims in case of spatial feature maps
-    feature_extractor = nn.Sequential(*layers)
+    try:
```
Comment (harshita-s) [inline InnerEye/ML/Histopathology/utils/layer_utils.py:19]: could we add a test for this function?

File: InnerEye/ML/Histopathology/models/deepmil.py (hunk 1 above)
Comment (dccastro) [reply InnerEye/ML/Histopathology/models/deepmil.py:122]: As we discussed, I'm moving dropout into `get_classifier()` and we can incorporate the hidden layer option in a future PR.

File: InnerEye/ML/Histopathology/utils/layer_utils.py (hunk 2 above)
Comment (dccastro) [reply InnerEye/ML/Histopathology/utils/layer_utils.py:19]: Added in `test_encoder.py`

**Review by harshita-s — DISMISSED** [review]

**Review by maxilse — APPROVED** [review]

File: InnerEye/ML/Histopathology/utils/layer_utils.py [hunk 3]
```diff
@@ -15,15 +15,23 @@ def get_imagenet_preprocessing() -> nn.Module:
...
+        # Otherwise fallback to sequence of child modules:
+        layers = list(pretrained_model.children())[:-1]
+        layers.append(nn.Flatten())  # flatten non-batch dims in case of spatial feature maps
+        feature_extractor = nn.Sequential(*layers)
+        with no_grad():
+            feature_shape = feature_extractor(rand(1, *input_dim)).shape
+        num_features = int(prod(as_tensor(feature_shape)).item())
     # fix weights, no fine-tuning
     for param in feature_extractor.parameters():
         param.requires_grad = False
```
Comment (harshita-s) [inline InnerEye/ML/Histopathology/utils/layer_utils.py:34]: This should not be set `False` for fine-tuning, right?
